src/components/location/location.service.ts

```diff
@@ -35,28 +33,21 @@ export class LocationService {
 
   async create(input: CreateLocation, session: Session): Promise<Location> {
     this.privileges.for(session, Location).verifyCan('create');
-    const checkName = await this.repo.doesNameExist(input.name);
-    if (checkName) {
-      throw new DuplicateException(
-        'location.name',
-        'Location with this name already exists.',
-      );
-    }
 
-    const { id, mapImageId } = await this.repo.create(input, session);
+    const dto = await this.repo.create(input, session);
     await this.files.createDefinedFile(
-      mapImageId,
+      dto.mapImage?.id,
```

Comment:
@CarsonF - I know we discussed this, and you mentioned that this whole thing needs an EdgeDB version or something like that.  Any idea how I can proceed until that happens?

Reply:
I'm not sure. It needs some more thought in the file layer before referencing here.

The big question is do files exist in DB without any backing versions.
## If Yes
Without any versions we cannot fulfill the GQL API, so we'd need to filter these out / convert to null in the API layer.
But having these would allow DefinedFiles to exist and work more or less the same.
They also wouldn't change IDs if all versions were deleted and then another uploaded, which seems like a good thing...but not really functionally breaking.
Created At would also be reset, even if deleting version and the uploading new version right after.
Still leaving potential leftovers in DB doesn't seem like the best thing.

## If No
Then we'd need to handle that File nullability.
Besides the opposite of the above, this mostly just changes what the abstraction does.
Creating file once vs as needed on new version upload.


I don't have any answers right now.

Reply:
I think we should closely match user intent. So we shouldn't create an empty file when the parent is created but rather on the first actual upload.

Resetting the ID, created at, and created by also rub me the wrong way.
We should keep these values for a potential future upload.

One solution might be to use a global and an access policy to automatically hide these empty files.
Idk if this would be more useful than an explicit filter.

I guess a third option would be to adapt the GQL API to handle these empty files.
Showing empty files in a directory list doesn't make sense. But saying a file of x id doesn't exist, but then allowing a version upload to that is also weird.
So perhaps exposing this distinction is the best way.

Reply:
In talking with Bryan, we decided it is hard & unintuitive to keep version-less files. So we'll delete it in this use case. Between that and creating defined files on demand, there will be no version-less files in EdgeDB.